registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java

```diff
@@ -728,4 +734,296 @@ public void testBioDeDupUpdatePacketHandlerProcessingFailWithAutoRejectEnable()
+        Mockito.when(utility.getUIn(anyString(), anyString(), any()))
+                .thenReturn("9403107396");
+        Mockito.when(idRepoService.getUinByRid(anyString(),any()))
+                .thenReturn("9403107397")
+                .thenReturn("9403107397")
+                .thenReturn("9403107397");
+
+        ReflectionTestUtils.setField(bioDedupeProcessor, "biometricsUpdateSingleMatchAutoReject", true);
+        ArgumentCaptor<InternalRegistrationStatusDto> argumentCaptor = ArgumentCaptor.forClass(InternalRegistrationStatusDto.class);
+        MessageDTO messageDto = bioDedupeProcessor.process(dto, stageName);
+        Mockito.verify(registrationStatusService).updateRegistrationStatus(argumentCaptor.capture(),anyString(),anyString());
+        assertFalse(messageDto.getIsValid());
+        assertFalse(messageDto.getInternalError());
+        Assert.assertEquals(argumentCaptor.getValue().getStatusCode(), RegistrationStatusCode.REJECTED.name());
+    }
+
+//    More than one match of RID/AID found at bio dedupe stage,
+//    all duplicate matches possess different UINs. Among that one UIN matches with update packet UIN
+    @Test
+    public void moreThenOneMatchFoundWithDiffrentUinOnlyOneMatches() throws ApisResourceAccessException, IOException,
+            PacketDecryptionFailureException, io.mosip.kernel.core.exception.IOException, JsonProcessingException, PacketManagerException {
+        registrationStatusDto.setRegistrationId("27847657360002520190320095011");
+        registrationStatusDto.setRegistrationType("UPDATE");
+        registrationStatusDto.setStatusCode("PROCESSING");
+        Mockito.when(registrationStatusService.getRegistrationStatus(any(),any(),any(), any())).thenReturn(registrationStatusDto);
+        Mockito.when(abisHandlerUtil.getPacketStatus(any()))
+                .thenReturn(AbisConstant.POST_ABIS_IDENTIFICATION)
+                .thenReturn(AbisConstant.DUPLICATE_FOR_SAME_TRANSACTION_ID);
+
+        Set<String> matchedRidList = new HashSet<>();
+        matchedRidList.add("27847657360002520190320095011");
+        matchedRidList.add("27847657360002520190320095012");
+        matchedRidList.add("27847657360002520190320095013");
+        matchedRidList.add("27847657360002520190320095014");
+        Mockito.when(abisHandlerUtil.getUniqueRegIds(any(), any(), anyInt(), any(), any())).thenReturn(matchedRidList);
+        Mockito.when(utility.getUIn(anyString(), anyString(), any()))
+                .thenReturn("9403107396");
+        Mockito.when(idRepoService.getUinByRid(anyString(),any()))
+                .thenReturn("9403107394")
+                .thenReturn("9403107395")
+                .thenReturn("9403107396");
+
+        ReflectionTestUtils.setField(bioDedupeProcessor, "biometricsUpdateSingleMatchAutoReject", false);
+        MessageDTO messageDto = bioDedupeProcessor.process(dto, stageName);
+        assertFalse(messageDto.getIsValid());
+        assertTrue(messageDto.getInternalError());
```

Comment:
We should use the ArgumentCaptor as well

Reply:
done